### ISSUE TITLE
Duplicates created when exporting/importing into/from calendar

### ISSUE THREAD

**Issue by gAQeNQ** [vmiklos/plees-tracker#150]
**Reproducer steps**
1. Log few sleeps
2. Export to calendar
3. Export again

**Actual result**
4. The calendar has duplicats
5. Same issue for import (duplicates are created in the Plees Tracker)

**Expected behavior**
Entry should have unique ID and skip duplicates when exporting/importing

**Comment by vmiklos** [comment]
@ed-george what do you think, what would be a good way to do this kind of sync? Assuming that the sleep id is lost on export, would the sleep start+end be a key when it comes to finding out if the sleep is already exported? Or remember what was the last exported sleep and continue from there? That might be cheaper.

**Comment by ed-george** [comment]
@vmiklos - That's a good question. I wonder if maybe exporting the sleep id to the event description would be an approach worth looking into or if the sleep id can be persisted to the event in another way. As far as I know, there's no mechanism for storing the id in the data structure of a calendar other than through fields the user could later change, so there's potentially no 'foolproof' solution to this.

Thinking slightly outside the box here; Maybe after the user's first successful export, when any subsequent export is requested, the user is presented with a dialog to allow them to export all sleeps SINCE their previous export or ALL sleeps.
Then the only thing that needs tracking is the time of the last sleep per export and no duplicate events are made.

**Comment by vmiklos** [comment]
Yes, I think simply remembering the timestamp of the last export is fair enough :-)

**Comment by gAQeNQ** [comment]
I have tested the feature and it looks that it works correctly. Thank you very much for fixing it.

**Comment by gAQeNQ** [comment]
Reading the conversation and the code of the PR this wont work in some cases. For example when I delete the app.

Lets assume this scenario:

1. I log some sleeps for some time
2. export them to calendar periodically so I have backup and visualisation
3. I change the phone
4. I install the app again
5. I do import from calendar
6. I do export
7. **I still get duplicates*
5.

**Comment by gAQeNQ** [comment]
I suggest to reopen this issue as the above is quite important IMO and it breaks a lot of data.

It is **VERY** painful to duplicate data when there is **NO** good way how to "undo" it. I have to delete whole calendar / clear app data and start all over again.

**Comment by vmiklos** [comment]
I've opened #209 instead. Reopening issues is a good way to destroy motivation. :-)
